python/triton/compiler/code_generator.py

```diff
@@ -583,7 +583,7 @@ def visit_then_else_blocks(self, node, liveins, then_block, else_block):
         ret_types = []
         ir_ret_types = []
         # variables in livein whose value is updated in `if`
-        for name in liveins:
+        for name in sorted(liveins):
```

Comment:
why does liveins has a different order run to run?

Reply:
I did not look deep at first and just ordered all dictionaries I found in this file.

After some additional investigation I found that root cause is on line 606: `then_defs.keys() & else_defs.keys()`
This statement is not stable, at least with python 3.9, which I am using.

Reply:
Seems sorting only this one set fixes the issue.
Should I limit this patch to one statement?

Reply:
oh interesting findings. Yes maybe only sorting the concatenation makes sense.

Reply:
done!

Not sure, does this change requires test?
This issues does not affect performance or functionality, it just annoying and makes compiler debugging harder.

Reply:
I agree